services/indexes/pvm/writer.go

```diff
@@ -209,52 +122,59 @@ type PtxDataModel struct {
 	ProposerType *string               `json:"proposerType,omitempty"`
 }
 
-func (w *Writer) ParseJSON(txBytes []byte) ([]byte, error) {
-	parsePlatformTx := func(b []byte) (*PtxDataModel, error) {
-		var block blocks.Block
-		_, err := w.codec.Unmarshal(b, &block)
-		if err != nil {
-			var blockTx txs.Tx
-			_, err = w.codec.Unmarshal(b, &blockTx)
-			if err != nil {
-				return nil, err
-			}
-			w.initCtxPtx(&blockTx)
-			txtype := reflect.TypeOf(&blockTx)
-			txtypeS := txtype.String()
-			return &PtxDataModel{
-				Tx:     &blockTx,
-				TxType: &txtypeS,
-			}, nil
-		}
-		w.initCtx(block)
-		blkID := ids.ID(hashing.ComputeHash256Array(b))
-		blkIDS := blkID.String()
-		btype := reflect.TypeOf(block)
+func (w *Writer) ParseJSON(b []byte) ([]byte, error) {
+	// Try and parse as a tx
+	tx, err := txs.Parse(blocks.GenesisCodec, b)
+	if err == nil {
+		tx.Unsigned.InitCtx(w.ctx)
+		txtype := reflect.TypeOf(tx) // Should this be `tx.Unsigned`?
```

Comment:
I think that would mean the current endpoint just returns junk (or always `SignedTx` or something)?

Reply:
Yeah - I think that's correct